rust/agama-network/src/dbus/tree_manager.rs

```diff
@@ -0,0 +1,164 @@
+use agama_lib::error::ServiceError;
+use parking_lot::Mutex;
+use uuid::Uuid;
+
+use crate::{dbus::interfaces, model::*, NetworkSystem};
+use std::collections::HashMap;
+use std::sync::Arc;
+
+/// Handle the objects in the D-Bus tree for the network state
+pub struct Tree {
+    connection: zbus::Connection,
+    network: Arc<Mutex<NetworkSystem>>,
+    objects: Arc<Mutex<ObjectsRegistry>>,
+}
+
+impl Tree {
+    pub fn new(connection: zbus::Connection, network: Arc<Mutex<NetworkSystem>>) -> Self {
+        Self {
+            connection,
+            network,
+            objects: Default::default(),
+        }
+    }
+
+    pub async fn populate(&mut self) -> Result<(), ServiceError> {
```

Comment:
I think here it would be useful to have documentation for this public method  as I kind of feel there will be some limitation

Reply:
Ouch, that was an old version of the `tree`. I have removed the file.